apps/cmp.c

```diff
@@ -0,0 +1,3404 @@
+    {"port", OPT_PORT, 's', "Act as HTTP mock server listening on given port"},
+    {"max_msgs", OPT_MAX_MSGS, 'n',
+     "max number of messages handled by HTTP mock server. Default: 0 = unlimited"},
+
+    {"srv_ref", OPT_SRV_REF, 's',
+     "Reference value to use as senderKID of server in case no -srv_cert is given"},
+    {"srv_secret", OPT_SRV_SECRET, 's',
+     "Password source for server authentication with a pre-shared key (secret)"},
+    {"srv_cert", OPT_SRV_CERT, 's', "Certificate used by the server"},
+    {"srv_key", OPT_SRV_KEY, 's',
+     "Private key used by the server for signing messages"},
+    {"srv_keypass", OPT_SRV_KEYPASS, 's',
+     "Server private key (and cert file) pass phrase source"},
+
+    {"srv_trusted", OPT_SRV_TRUSTED, 's',
+     "Trusted certificates for client authentication"},
+    {"srv_untrusted", OPT_SRV_UNTRUSTED, 's',
+     "Intermediate certs that may be useful for verifying CMP protection"},
+    {"rsp_cert", OPT_RSP_CERT, 's',
+     "Certificate to be returned as mock enrollment result"},
+    {"rsp_extracerts", OPT_RSP_EXTRACERTS, 's',
+     "Extra certificates to be included in mock certification responses"},
+    {"rsp_capubs", OPT_RSP_CAPUBS, 's',
+     "CA certificates to be included in mock ip response"},
+    {"poll_count", OPT_POLL_COUNT, 'n',
+     "Number of times the client must poll before receiving a certificate"},
+    {"check_after", OPT_CHECK_AFTER, 'n',
+     "The check_after value (time to wait) to include in poll response"},
+    {"grant_implicitconf", OPT_GRANT_IMPLICITCONF, '-',
+     "Grant implicit confirmation of newly enrolled certificate"},
+
+    {"pkistatus", OPT_PKISTATUS, 'n',
+     "PKIStatus to be included in server response. Possible values: 0..6"},
+    {"failure", OPT_FAILURE, 'n',
+     "A single failure info bit number to include in server response, 0..26"},
+    {"failurebits", OPT_FAILUREBITS, 'n',
+     "Number representing failure bits to include in server response, 0..2^27 - 1"},
+    {"statusstring", OPT_STATUSSTRING, 's',
+     "Status string to be included in server response"},
+    {"send_error", OPT_SEND_ERROR, '-',
+     "Force server to reply with error message"},
+    {"send_unprotected", OPT_SEND_UNPROTECTED, '-',
+     "Send response messages without CMP-level protection"},
+    {"send_unprot_err", OPT_SEND_UNPROT_ERR, '-',
+     "In case of negative responses, server shall send unprotected error messages,"},
+    {OPT_MORE_STR, 0, 0,
+     "certificate responses (ip/cp/kup), and revocation responses (rp)."},
+    {OPT_MORE_STR, 0, 0,
+     "WARNING: This setting leads to behavior violating RFC 4210"},
+    {"accept_unprotected", OPT_ACCEPT_UNPROTECTED, '-',
+     "Accept missing or invalid protection of requests"},
+    {"accept_unprot_err", OPT_ACCEPT_UNPROT_ERR, '-',
+     "Accept unprotected error messages from client"},
+    {"accept_raverified", OPT_ACCEPT_RAVERIFIED, '-',
+     "Accept RAVERIFED as proof-of-possession (POPO)"},
```

Comment:
Typo: RAVERIF**I**ED

Reply:
Thx for pointing out these nits; fixed typo.